core/types/block_ext_test.go

```diff
@@ -112,108 +115,194 @@ func assertDifferentPointers[T any](t *testing.T, a *T, b any) {
-		Version: 1,
+	// Golden data from original coreth implementation, before integration of
+	// libevm. WARNING: changing these values can break backwards compatibility
+	// with extreme consequences.
+	const wantHex = "f90235dedd0105049402000000000000000000000000000000000000000306808080f90211f9020ea00900000000000000000000000000000000000000000000000000000000000000a00000000000000000000000000000000000000000000000000000000000000000940000000000000000000000000000000000000000a00000000000000000000000000000000000000000000000000000000000000000a00000000000000000000000000000000000000000000000000000000000000000a00000000000000000000000000000000000000000000000000000000000000000b9010000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000070880808080a00000000000000000000000000000000000000000000000000000000000000000880000000000000000a00a000000000000000000000000000000000000000000000000000000000000000c0d"
+
+	assert.Equal(t, wantHex, hex.EncodeToString(encoded), "golden data")
+}
+
+// blockWithNonZeroFields returns a [Block] and a [BlockBodyExtra],
+// each with all fields set to non-zero values.
+// The [BlockBodyExtra] extra payload is set in the [Block] via `extras.Block.Set`.
+//
+// NOTE: They can be used to demonstrate that RLP round-trip encoding
+// can recover all fields, but not that the encoded format is correct. This is
+// very important as the RLP encoding of a [Block] defines its hash.
+func blockWithNonZeroFields() (*Block, *BlockBodyExtra) {
+	header := &Header{
+		ParentHash: common.Hash{1},
+	}
+	headerExtra := &HeaderExtra{
+		ExtDataHash: common.Hash{2},
+	}
+	SetHeaderExtra(header, headerExtra)
+
+	tx := NewTransaction(1, common.Address{2}, big.NewInt(3), 4, big.NewInt(5), []byte{6})
+	txs := []*Transaction{tx}
+
+	uncle := &Header{
+		Difficulty: big.NewInt(7),
+		Number:     big.NewInt(8),
+		ParentHash: common.Hash{9},
+	}
+	uncleExtra := &HeaderExtra{ExtDataHash: common.Hash{10}}
+	SetHeaderExtra(uncle, uncleExtra)
+	uncles := []*Header{uncle}
+
+	receipts := []*Receipt{{PostState: []byte{11}}}
+
+	block := NewBlock(header, txs, uncles, receipts, stubHasher{})
+	extra := &BlockBodyExtra{
+		Version: 12,
+		ExtData: &[]byte{13},
 	}
-	assert.Equal(t, wantExtra, extra)
+	extras.Block.Set(block, extra)
+	return block, extra
 }
 
-func TestBodyExtraGetWith(t *testing.T) {
+func TestBlockWithNonZeroFields(t *testing.T) {
 	t.Parallel()
 
-	b := &Body{}
+	block, extra := blockWithNonZeroFields()
+	t.Run("Block", func(t *testing.T) { allExportedFieldsSet(t, block, "ReceivedAt", "ReceivedFrom") })
```

Comment:
nit: maybe 
```suggestion
	allowedExceptions := []string{"ReceivedAt", "ReceivedFrom")
	t.Run("Block", func(t *testing.T) { allExportedFieldsSet(t, block, allowedExceptions...) })
```

Reply:
I agree. I could infer what they meant but there's a tiny bit of uncertainty without the variable explaining it.